code/modules/clothing/clothing.dm

```diff
@@ -355,6 +355,16 @@ BLIND     // can't see anything
 		flash_protect ^= initial(flash_protect)
 	if(visor_vars_to_toggle & VISOR_TINT)
 		tint ^= initial(tint)
+		
+/obj/item/clothing/head/helmet/space/plasmaman/security/visor_toggling() //security envirohelmets get flash protection without visor
+	up = !up
+	clothing_flags ^= visor_flags
+	flags_inv ^= visor_flags_inv
+	icon_state = "[initial(icon_state)]"
+	if(visor_vars_to_toggle & VISOR_FLASHPROTECT)
+		flash_protect = initial(flash_protect) - up
+	if(visor_vars_to_toggle & VISOR_TINT)
+		tint ^= initial(tint)
```

Comment:
Flash protection should come from glasses not the helmet itself due to free flash protection for getting the helmet. Description of the helmet should be updated to remove the flash protection instead.